src/Core/Managed/Shared/Extensibility/OperationCorrelationTelemetryInitializer.cs

```diff
@@ -22,27 +28,86 @@ public void Initialize(ITelemetry telemetryItem)
         {
             var itemContext = telemetryItem.Context.Operation;
 
-            if (string.IsNullOrEmpty(itemContext.ParentId) || string.IsNullOrEmpty(itemContext.Id) || string.IsNullOrEmpty(itemContext.Name))
+            bool isActivityEnabled = false;
+#if !NET40
+            isActivityEnabled = ActivityProxy.TryRun(() =>
             {
-                var parentContext = CallContextHelpers.GetCurrentOperationContext();
-                if (parentContext != null)
+                if (Activity.Current == null)
+                {
+                    return false;
+                }
+
+                var currentActivity = Activity.Current;
```

Comment:
Move this up above line 35 and use it for the null check. That way you access the ```.Current``` property just once. It'll be cheaper (since ```Activity.Current``` isn't simply a static field).